tests/bcl-test/BCLTests/RegisterType.cs.in

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.IO;
+using System.Collections.Generic;
+%USING%
+
+namespace Xamarin.iOS.UnitTests {
+	// autogenerated class that uses types from the test assemblies so that
+	// they are aot and not linked away.
+	public static class RegisterType
+	{
+		public static Dictionary <string, Type> TypesToRegister = new Dictionary <string, Type> {
+%KEY VALUES%
+		};
+
+		public static void RegisterTypes () 
+		{
+			foreach (var a in TypesToRegister.Keys) {
+				// do something with the type, so that it is not removed
+				var assemblyPath = Path.GetFileName (TypesToRegister[a].Assembly.Location);
+				Console.WriteLine ($"We hare using type {TypesToRegister[a]} from assembly {assemblyPath}");
```

Comment:
`hare` ? 🐰 
or `are` 😄

Reply:
lol are are

Reply:
Done, not push not to abort build in bots.

Reply:
Fixed.